src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -792,6 +792,14 @@ void UserInfoPopup::updateUserData()
             return;
         }
 
+        // Correct for when being opened with "id:"
+        if (this->userName_ != user.login)
+        {
+            this->userName_ = user.login;
+
+            // Ensure recent messages are shown
+            this->updateLatestMessages();
```

Comment:
`updateLatestMessages()` will run twice now, once with #id then with correct username from helix.
Would be best to avoid that, maybe prefix detection could be done sooner in `setData()` and just pass bool to `updateUserData()`

Reply:
I considered two options:
 - rewrite the message filter to allow for checking for IDs
 - just call `updateLatestMessages()` when the information is available, which seemed fine until I just saw that it creates a new signal connection.

I'll make it skip the first call.

Reply:
Updated in [ba4151b](https://github.com/Chatterino/chatterino2/pull/4934/commits/ba4151b774a73424980ab4e918076fa2fb40b2e7)